dart/lib/src/sentry.dart

```diff
@@ -316,6 +316,8 @@ class Sentry {
   static ISentrySpan? getSpan() => _hub.getSpan();
 
   /// Gets access to the metrics API for the current hub.
+  @Deprecated(
+      'Metrics will be deprecated and removed in the next major release. Sentry will reject all metrics sent after October 7, 2024. Learn more: https://sentry.zendesk.com/hc/en-us/articles/26369339769883-Upcoming-API-Changes-to-Metrics')
```

Comment:
`m`: Aren't metrics experimental? If yes, we can also remove them in a nonmajor update. I don't think it makes sense to keep the code if, anyways, we are going to reject all metrics after October 7th.

Reply:
Hmm, I don't have any strong opinion on that. 

@kahest what do you think?

Reply:
I'd deprecate now and leave some time to remove - not because it needs to be a major (can be a minor in a few weeks), but to give app maintainers at least a bit of time to learn about the deprecation and remove the code before we break their builds. even if it's likely a very small change they need to do, we don't need to force them to do it right now

Reply:
Sounds good, I guess we can approve this PR 

I'd like to bring that in for the release today